OpenKeychain/src/main/java/org/sufficientlysecure/keychain/pgp/PgpKeyOperation.java

```diff
@@ -1192,81 +1139,38 @@ private PgpEditKeyResult internalRestricted(PGPSecretKeyRing sKR, SaveKeyringPar
+    public PgpEditKeyResult addS2kToSubKeys(CanonicalizedSecretKeyRing wsKR,
+                                            Passphrase newPassphrase) {
 
         OperationLog log = new OperationLog();
         int indent = 0;
 
-        if (changeUnlockParcel.mMasterKeyId == null || changeUnlockParcel.mMasterKeyId != wsKR.getMasterKeyId()) {
-            log.add(LogType.MSG_MF_ERROR_KEYID, indent);
-            return new PgpEditKeyResult(PgpEditKeyResult.RESULT_ERROR, log, null);
-        }
-
         log.add(LogType.MSG_MF, indent,
                 KeyFormattingUtils.convertKeyIdToHex(wsKR.getMasterKeyId()));
         indent += 1;
         progress(R.string.progress_building_key, 0);
 
-        // We work on bouncycastle object level here
         PGPSecretKeyRing sKR = wsKR.getRing();
         PGPSecretKey masterSecretKey = sKR.getSecretKey();
         PGPPublicKey masterPublicKey = masterSecretKey.getPublicKey();
-        // Make sure the fingerprint matches
-        if (changeUnlockParcel.mFingerprint == null || !Arrays.equals(changeUnlockParcel.mFingerprint,
-                masterSecretKey.getPublicKey().getFingerprint())) {
-            log.add(LogType.MSG_MF_ERROR_FINGERPRINT, indent);
-            return new PgpEditKeyResult(PgpEditKeyResult.RESULT_ERROR, log, null);
-        }
-
-        // Find the first unstripped secret key
-        PGPSecretKey nonDummy = firstNonDummySecretKeyID(sKR);
-        if(nonDummy == null) {
-            log.add(OperationResult.LogType.MSG_MF_ERROR_ALL_KEYS_STRIPPED, indent);
-            return new PgpEditKeyResult(PgpEditKeyResult.RESULT_ERROR, log, null);
-        }
-
-        if (!cryptoInput.hasPassphrase()) {
-            log.add(LogType.MSG_MF_REQUIRE_PASSPHRASE, indent);
 
-            return new PgpEditKeyResult(log, RequiredInputParcel.createRequiredSignPassphrase(
-                    masterSecretKey.getKeyID(), nonDummy.getKeyID(),
-                    cryptoInput.getSignatureTime()), cryptoInput);
-        } else {
-            progress(R.string.progress_modify_passphrase, 50);
-            log.add(LogType.MSG_MF_PASSPHRASE, indent);
-            indent += 1;
+        log.add(LogType.MSG_MF_ADD_S2K_ENCRYPTION, indent);
+        indent += 1;
 
-            try {
-                sKR = applyNewPassphrase(sKR, masterPublicKey, cryptoInput.getPassphrase(),
-                        changeUnlockParcel.mNewPassphrase, log, indent);
-                if (sKR == null) {
-                    // The error has been logged above, just return a bad state
-                    return new PgpEditKeyResult(PgpEditKeyResult.RESULT_ERROR, log, null);
-                }
-            } catch (PGPException e) {
-                throw new UnsupportedOperationException("Failed to build encryptor/decryptor!");
+        try {
+            sKR = applyNewPassphrase(sKR, masterPublicKey, new Passphrase(), newPassphrase, log, indent);
```

Comment:
Yep, thought that it'll be better to just stick to the convention already present.

It seems that we use pretty strange naming in some tests as well. 
mXXX for static variables. Left that alone for the time being.